### ISSUE TITLE
I just want to confirm if output is correct or not

### ISSUE THREAD

**Issue by Xuquansheng** [bbycroft/llm-viz#13]
as mentioned, Its goal is a simple one: take a sequence of six letters:
C B A B B C

and sort them in alphabetical order, i.e. to "ABBBCC". but output is CBAAAA
![火狐截图_2023-12-16T13-10-25 610Z](https://github.com/bbycroft/llm-viz/assets/82086188/c264651b-ed40-4c31-a4d3-bdeeb467b1ed)

**Comment by YoungHuang** [comment]
Same question.

**Comment by www222fff** [comment]
also make me confused. Please clarify

**Comment by fredbenenson** [comment]
I have this question as well – the output is sorted.. just backwards but without the correct number of characters (e.g. there should be 2 C's, etc.

**Comment by sampath1994** [comment]
yeah, same question

**Comment by bbycroft** [comment]
Yeah it's not entirely clear here, although the output is correct.

The "output" is from the 6th column onwards, i.e. the less-gray 'A' is the first of the real sorted output.

A given output column can only "see" the inputs from its own and preceding column via attention, so we don't expect the first 6 columns to match the sorted value of those same columns.

**Comment by NightNightNight** [comment]
I would like to know what the meaning of the blank is after the input token.
<img width="1846" height="678" alt="Image" src="https://github.com/user-attachments/assets/d07b44b8-d739-4818-a8c6-38d1fa06982f" />
